optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -1100,3 +1107,811 @@ def _raise_invalid_batch_size(
+            "sample": sample,
+            "timestep": timestep,
+            "encoder_hidden_states": encoder_hidden_states,
+            "mid_block_additional_residual": down_and_mid_block_samples[-1],
+        }
+        a = 1
+        for block in down_and_mid_block_samples:
+            if a == 23:
+                inputs["down_block_additional_residual"] = block
+                break
+            else:
+                inputs[f"down_block_additional_residual_{a}"] = block
+            a += 2
+
+        if text_embeds is not None:
+            inputs["text_embeds"] = text_embeds
+        if time_ids is not None:
+            inputs["time_ids"] = time_ids
+        if timestep_cond is not None:
+            inputs["timestep_cond"] = timestep_cond
+
+        outputs = self.request(inputs, share_inputs=True)
+        return list(outputs.values())
+
+
+class OVStableDiffusionControlNetPipelineBase(OVStableDiffusionPipelineBase):
+    """
+    OpenVINO inference pipeline for Stable Diffusion with ControlNet guidence
+    """
+
+    auto_model_class = StableDiffusionControlNetPipeline
+    export_feature = "stable-diffusion-controlnet"
+
+    def __init__(
+        self,
+        unet: openvino.runtime.Model,
+        controlnet: openvino.runtime.Model,
+        scheduler: Union[None, "DDIMScheduler", "PNDMScheduler", "LMSDiscreteScheduler"],
+        text_encoder: Optional[openvino.runtime.Model] = None,
+        text_encoder_2: Optional[openvino.runtime.Model] = None,
+        vae_decoder: Optional[openvino.runtime.Model] = None,
+        vae_encoder: Optional[openvino.runtime.Model] = None,
+        tokenizer: Optional["CLIPTokenizer"] = None,
+        tokenizer_2: Optional["CLIPTokenizer"] = None,
+        feature_extractor: Optional["CLIPFeatureExtractor"] = None,
+        device: str = "CPU",
+        dynamic_shapes: bool = False,
+        compile: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        quantization_config: Optional[Union[OVWeightQuantizationConfig, Dict]] = None,
+        **kwargs,
+    ):
+        # super().__init__()
+        self._device = device.upper()
+        self.ov_config = {} if ov_config is None else {**ov_config}
+        self.is_dynamic = dynamic_shapes
+        self.preprocessors = []
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        self._model_save_dir_tempdirectory_instance = None
+        if isinstance(model_save_dir, TemporaryDirectory):
+            self._model_save_dir_tempdirectory_instance = model_save_dir
+            self._model_save_dir = Path(model_save_dir.name)
+        elif isinstance(model_save_dir, str):
+            self._model_save_dir = Path(model_save_dir)
+        else:
+            self._model_save_dir = model_save_dir
```

Comment:
Please apply same fix for controlnet in export_controlnet(): https://github.com/huggingface/optimum-intel/pull/749